### ISSUE TITLE
Not seeing this working in HTML file

### ISSUE THREAD

**Issue by fuzenco** [alexanderweiss/nova-prettier#5]
Running Prettier in Nova doesn’t seem to do anything. For instance, just as a test, running this in an HTML file show the following code untouched as far as indention…

```
<div id="error-message"></div>
  
  <p>Testing this.</p>
    
    <p>Is this working?</p>
```

You’d think it would outdent those 3 lines to be equal (flush left) with the div tag.

**Comment by fuzenco** [comment]
FYI: The other Prettier extension in Nova’s Extension gallery operates the same way, so perhaps it’s a Nova issue?

**Comment by alexanderweiss** [comment]
Sorry for the late reply, but are you still running into this? And do you have a `Format using Prettier` option under the `Editor` menu that's enabled?

**Comment by mattdarveniza** [comment]
I'm also seeing this extension do nothing to expected files, not only in html but in anything (js, ts, etc).

The Editor menu is enabled for me, but formatting just does nothing.

Looked in the logs and this seems to be why:
```
Prettier[15:19:46.709000] Unable to find or install prettier Error: env: npm: No such file or directory

Prettier[15:19:46.755000] Error trying to find workspace Prettier env: npm: No such file or directory


Prettier[15:19:46.756000] Using prettier from /Users/matt/Library/Application Support/Nova/Extensions/alexanderweiss.prettier/node_modules/prettier (extension located at: /Users/matt/Library/Application Support/Nova/Extensions/alexanderweiss.prettier)

Prettier[15:19:46.758000] Unable to set up prettier service Error: Could not find a module at path: “../../../../../../../../Users/matt/Library/Application Support/Nova/Extensions/alexanderweiss.prettier/node_modules/prettier/standalone.js”
```

`npm` is definitely on my path, but seemingly however Nova does scripting isn't working? Not sure if a Nova issue or a problem with this extension.

**Comment by alexanderweiss** [comment]
Thanks for the logs @mattdarveniza! Do you use `nvm` by any chance?

**Comment by mattdarveniza** [comment]
@alexanderweiss yes I use nvm. Now that you say that I can immediately see why that's a problem 😅 . Presumably Nova won't run the bash profile where those aliases to `npm` are set up.

Is that something resolvable by the extension or something I should bring up with the Nova devs?

**Comment by alexanderweiss** [comment]
Yes, I think `env` does use your `PATH` but won’t run the script nvm. Though I need to look into how that works exactly.

This all happens in the extension (and a package by one of the people at Panic), so it shouldn’t be a problem to fix it and create an issue/PR in the package’s repo. I’ll look at it in the coming week, but you’re also welcome to create a PR of course. It’d be great if you could test it in any case.

**Comment by GeorgeWS** [comment]
Ahh, same issue here (Prettier extension not doing anything when saving or clicking "Format with Prettier")—and, similar/same cause: I'm using `nodenv` to manage node versions, so presumably its shims aren't showing up in the `PATH` when Prettier is invoked.

**Comment by mattdarveniza** [comment]
For what it's worth, I resolved this for now by just installing node through brew. Since nodenv and nvm both apply shims over the path, this didn't interfere with any of my other tools and got this extension working just fine. What's one more installation of node, right? 😉

**Comment by fuzenco** [comment]
Thanks @mattdarveniza . I’d actually deleted this extension but I might try your suggestion.

**Comment by alexanderweiss** [comment]
Panic's suggested solution is to move the NVM configuration to your login shell configuration: https://library.panic.com/nova/environment-variables/ (their comment: https://dev.panic.com/panic/nova-issues/-/issues/1488#note_7395).

I've updated this extension to show a UI warning when NPM seems to be available, pointing you to that documentation. It's not perfect, but I think that should solve it for now?

**Comment by cedricdelpoux** [comment]
Moving the following from `.zshrc` to `~/.bash_profile` worked for me but it's not normal. Prettier is working without this fix in *Atom* or *VSCode*
```
export NVM_DIR=~/.nvm
source $(brew --prefix nvm)/nvm.sh
```

**Comment by alexanderweiss** [comment]
@cedricdelpoux I agree that it's a little annoying. But unlike both of those editors Nova doesn't come with its own Node runtime. So the extension either has to install Node itself or use whatever Node you have installed. I've opted for the latter. But since Nova starts login shells for extension, no interactive shells, you have to configure NVM in a login shell if you don't have Node installed separately from NVM. The extension could possibly start a login shell, but that may cause unintended side-effects (the reason Nova doesn't do it by default).

**Comment by cedricdelpoux** [comment]
Thank you for you explanation
